src/routes/safe/components/Apps/components/AppFrame.tsx

```diff
@@ -170,6 +171,14 @@ const AppFrame = ({ appUrl }: Props): ReactElement => {
       chainId: NETWORK_ID,
     }))
 
+    communicator?.on('getSafeBalances', async (msg) => {
+      const { currency = 'usd' } = msg.data.params as GetBalanceParams
```

Comment:
There's a Redux selector `currentCurrencySelector` if you want to take user's currency as the default.

Reply:
I thought that this is unobvious behavior and decided not to do that. 

In the discussion above Richard suggested to returned cached data from redux but it has a problem similar to using the user's selected currency:

> Also, the way it's implemented right now (useEffect hook), usage of state piece (balances) would mean the communicator would mount/unmount on state changes, which could lead to some app's messages being lost. I wanted to rewrite it in a way so communicator would spawn only once but haven't got the chance to do that yet.